bson/value_reader_test.go

```diff
@@ -779,6 +870,7 @@ func TestValueReader(t *testing.T) {
 				(*valueReader).ReadString,
 				"",
 				io.ErrUnexpectedEOF,
+				io.EOF,
 				TypeString,
```

Comment:
Is this the only case where the error differs between the streaming and buffered solutions?

Optional: Consider updating the logic to always return `io.EOF`. That should allow removing `streamingErr` and `bufferedError` from the test cases.